rotkehlchen/accounting/rules.py

```diff
@@ -105,42 +25,49 @@ def __init__(
         self.aggregators = evm_aggregators
         self.pot = pot
         self.event_settings: dict[int, BaseEventSettings] = {}
+        self.event_callbacks: dict[int, EventsAccountantCallback] = {}
 
     def _query_db_rules(self) -> None:
         """Query the accounting rules in the db and update event_settings with them"""
-        with self.database.conn.read_ctx() as cursor:
-            cursor.execute(
-                'SELECT type, subtype, counterparty, taxable, count_entire_amount_spend, '
-                'count_cost_basis_pnl, accounting_treatment FROM accounting_rules',
+        rules_info, _ = DBAccountingRules(self.database).query_rules(
+            filter_query=AccountingRulesFilterQuery.make(),
+        )
+        for rule_info in rules_info:
+            key = get_event_type_identifier(
+                event_type=rule_info.event_key[0],
+                event_subtype=rule_info.event_key[1],
+                counterparty=rule_info.event_key[2],
             )
-            for entry in cursor:
-                rule = BaseEventSettings.deserialize_from_db(entry[3:])
-                key = get_event_type_identifier(
-                    event_type=HistoryEventType.deserialize(entry[0]),
-                    event_subtype=HistoryEventSubType.deserialize(entry[1]),
-                    counterparty=entry[2] if entry[2] != NO_ACCOUNTING_COUNTERPARTY else None,
-                )
-                self.event_settings[key] = rule
+            self.event_settings[key] = rule_info.rule
 
-    def get_event_settings(self, event: HistoryBaseEntry) -> Optional[BaseEventSettings]:
-        """Return a matching rule for the event if it exists"""
-        rule = self.event_settings.get(event.get_type_identifier(), None)
+    def get_event_settings(
+            self,
+            event: HistoryBaseEntry,
+    ) -> tuple[Optional[BaseEventSettings], Optional[EventsAccountantCallback]]:
+        """
+        Return a matching rule for the event if it exists and a optional callback defined for
+        the rule that should be executed
+        """
+        event_identifier = event.get_type_identifier()
+        rule = self.event_settings.get(event_identifier, None)
         if isinstance(event, EvmEvent) is False or rule is not None:
-            return rule
+            return rule, self.event_callbacks.get(event_identifier)
 
-        return self.event_settings.get(event.get_type_identifier(include_counterparty=False), None)
+        event_identifier = event.get_type_identifier(include_counterparty=False)
+        return (
```

Comment:
What happens if there is no rule for the event?

Reply:
If there is no rule then this function returns None and the event will be skipped

Reply:
Can we have a log entry for these cases? To let client know something was skipped?